ibis/expr/operations/relations.py

```diff
@@ -395,7 +395,7 @@ def name(self) -> str:
             return f"{name}_ref"
         return util.gen_name("self_ref")
 
-    @property
+    @attribute
```

Comment:
This shouldn't contribute to the call stack, the calculated properties are not compared in `Concrete` subclasses just the `__args__`.

Reply:
They don't directly contribute to the call stack, but the traversal (at least for binary relational operations) O(N ** 2) at the moment.

Reply:
Attributes are not traversed either.

Reply:
**Something** is accessing `schema` during `left.equals(right)` so I'm not sure what you mean by "attributes aren't traversed"